File: engine/engine-model/src/main/scala/tools/packet/movement/MoveDragon.scala
package tools.packet.movement

import java.awt.Point

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MoveDragon(private var _ownerId: Int, private var _startPosition: Point, private var _movementList: java.util.List[java.lang.Byte]) extends PacketInput {
  def ownerId: Int = _ownerId

  def startPosition: Point = _startPosition

  def movementList: java.util.List[java.lang.Byte] = _movementList

  override def opcode(): SendOpcode = SendOpcode.MOVE_DRAGON
}

File: engine/engine-model/src/main/java/tools/packet/movement/MoveDragon.java
package tools.packet.movement; 
import java.awt.Point;
import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MoveDragon(Integer ownerId, Point startPosition, List<Byte> movementList) implements PacketInput {
  @Override
  public SendOpcode opcode() {
    return SendOpcode.MOVE_DRAGON;
  }
}